### ISSUE TITLE
[suggestion] warp rework

### ISSUE THREAD

**Issue by king32109** [tsunamayo/Starship-EVO#6152]
### Short Description of the issue:
i was thinking as or recent that warp needed some work, few issues I have with it currently, warp is to accessible, you can just fly to a planet besides the starter planet and void pears are everywhere, i find that is too easy to get to warp your ship, second issue is how they are distributed or the lack thereof, now they spawn randomly all over the planet, they make it also easy to warp very soon, what i also don’t like is for large ships that are bigger like 200m or higher needing a ridiculous amount of pearls, an empty hull I have that is 380m is needing 500 pearls for the 10 jumps, which is insane, sure you might not build that big in survival but if we get a way to transfers creative build ships into survival it becomes viable in a way to get those ships into survival, especially easy for multiplayer is easier to get since you can then make many players mine for the resources. 

so my suggestions for these issues are as followed.
1: make void pearls spawn in patches and not all around the planet, this would fix the random look of the pearls spawning, and it would make it also to go searching and adventuring the planet more, if more gets added you want more incentive to look for pearls and events, now you don’t have that, i suspect that is coming planet side events but the pearls being more rare and patch would make it more balanced and make looking for them a necessity and part of the game play loop

2: pearl rarities, this i think can be done 2 ways, one i though of is making a pearl that is different, let’s make it purple, the purple pearl would let say do the same as 10 0.5m pearls, but are harder to find, if you have a patch (of idea number 1) of let’s say 30 pearls, you would find 1 to 5 of the rare purple pearls, this would fix the large ship issue, where they need more then you find reasonable, if you have one of these pearls they count as 10, but they would also work with normal pearls, like if you have already, 10 blue pearls, and 1 purple pearls, you can jump 20 times, of course the larger the ship the more you need, but this would require Biggers ships to not need as many pearls then they do now, most people i know on discord like to build big ships, only accounting for smaller ships is not a good way to develop the warp system. 
the second way to make pearls more viable for bigger ships would be a new block, one that allows you to compress pearls into more efficient pearls, this block would take let’s say 5 normal pearls and compress them into a new pearl, but the new one would be 5 times as efficient, so 1 of those pearls would make it so you can jump 5 times or possible more, if you make addon modules to make the compressor even more efficient, 

these 2 ideas would make it easier to warp for larger ships and make warp accessible but not to much as it is now, it also adds a new mechanic in the compressor to make factory buildings more diverse.

**Comment by najlitarvan** [comment]
having pearls in patches could also allow for increased and believable interactions, since we could have ai both friendly and agresive around the patches, depending how it could work, there could be rare patches that could slowly regenerate pearls, giving a player why to explore the whole planet, to find more of these patches

**Comment by CoasterBlock** [comment]
I voiced similar concernces in #5973. Though, Id argue, void pearls shouldnt spawn on planets at all. Instead Id say that they should spawn far away from all planets on the edges of solar systems. In patches or clouds. Or you would harvest them from some other stellar object for which youd need to build an actual strcture for. I mean, we can already build sizable structures. May as well make building "mandatory" for progression. Have players build some kind of construct with wich to generate/harvest void pearls at a high energy cost.

**Comment by CoasterBlock** [comment]
I voiced similar concernces in #5973. Though, Id argue, void pearls shouldnt spawn on planets at all. Instead Id say that they should spawn far away from all planets on the edges of solar systems. In patches or clouds. Or you would harvest them from some other stellar object for which youd need to build an actual strcture for. I mean, we can already build sizable structures. May as well make building "mandatory" for progression. Have players build some kind of construct with wich to generate/harvest void pearls at a high energy cost.

**Comment by najlitarvan** [comment]
@CoasterBlock honestly some kind of hybrid would make sense, small quantities shouldn't need builds, but for bigger quantities? absolutely, would also turn big ships from impossibilities and into a showcase of logistics setup

**Comment by tsunamayo** [comment]
Hi, yes I will improve the current generation I agree they are too common. For now I am not too worried about 380m ship, they will be hard to make in survival anyway (you would need to mine everything). 500 pearls is not a big deal if you can use the collector to get them and transfert them automatically (which I still need to do).

**Comment by king32109** [comment]
True they might be hard. But for me i like to build stuff that csn function in survival. If you only focus on small stuff people wont do what this game has to offer

**Comment by CoasterBlock** [comment]
I agree with the points mentioned. It should be viable perhaps even necessary to make larger structures in the game after lets say the mid game. a 300m ship shouldnt be large, it should be the standard. Players should have the ability to make "megastructures" in survival. Because its a space game, mining everything may actually be plausible. Build a set of drones to harvest a ring of a planet maybe. 
In addition id argue void pearls should be routable using cargo-pipes. Larger ships will need larger warp cores and this means, warp pearls need to be stored somewhere.

**Comment by thriple** [comment]
Why not make it a resource node or add resource nodes alongside patches? Set different material requirements for different size mining lasers so we're not pulling too much, too soon. Set up a mining post. Gives us a reason to make a base and return to it, upgrade it.
